hooks/pre-commit

```diff
@@ -0,0 +1,32 @@
+#!/bin/sh -x
+#
+# Recommended pre-commit git hook for activity-stream github repo
+#
+# Install by executing
+#
+#   ln -s ../../hooks/pre-commit .git/hooks/pre-commit
+#
+# at the top-level of the activity-stream github repo.
+#
+# Runs `eslint --fix` on all selected files, which, given our current
+# prettier configuration, means prettifying these files as well.  The
+# commit will be aborted if eslint exits with a failure code.
+#
+# Based on the example script in the prettier docs at
+# https://prettier.io/docs/en/precommit.html
+
+FILES=$(git diff --cached --name-only --diff-filter=ACM "*.js" "*.jsx" "*.jsm" | sed 's| |\\ |g')
+echo "files to check: $FILES"
+[ -z "$FILES" ] && exit 0
+
+echo "$FILES" | xargs -t ./node_modules/.bin/eslint --fix
+if [ $? -ne 0 ]
+then
+  echo "eslint failed; aborting commit.  Please fix and retry."
+  exit -1
```

Comment:
I thought we said we wanted it to just continue even on failure? Or was that only in the context of rebases? The flow for a regular commit wouldn't really be that much different even if we continue on failure. Instead of plain retry commit, it would be commit amend or a new commit to squash later.

Reply:
As per the standup, I'm going to leave the abort in place.  If this really turns out to be a problem, we'll change it.